contracts/service-registry/src/state.rs

```diff
@@ -195,6 +214,64 @@ pub fn remove_service_override(
     Ok(())
 }
 
+pub fn update_verifier_authorization_status(
+    storage: &mut dyn Storage,
+    service_name: &ServiceName,
+    auth_state: &AuthorizationState,
+    verifiers: &[Addr],
+) -> error_stack::Result<(), ContractError> {
+    let mut authorized_count_change = 0i32;
+
+    for verifier_addr in verifiers.iter() {
+        VERIFIERS
+            .update(
+                storage,
+                (service_name, verifier_addr),
+                |existing_verifier| -> std::result::Result<Verifier, ContractError> {
+                    let previous_state = existing_verifier
+                        .as_ref()
+                        .map(|verifier| verifier.authorization_state.clone());
+
+                    update_auth_verifier_count_change_tracker(
+                        &mut authorized_count_change,
+                        &previous_state,
+                        auth_state,
+                    )?;
```

Comment:
```suggestion
                    authorized_count_change = authorized_count_change.checked_add(calculate_auth_verifier_count_change(previous_state.as_ref(), &auth_state)).ok_or(ContractError::AuthorizedVerifiersIntegerOverflow);
```
changes according to my other suggestion.